File: backend/src/utils/chunker.js
import { RecursiveCharacterTextSplitter } from "@langchain/textsplitters";
export async function chunkText(text) {
  const splitter = new RecursiveCharacterTextSplitter({
    chunkSize: 500,
    chunkOverlap: 100,
    separators: ["\n\n", "\n", " ", ""]
  });

  const docs = await splitter.createDocuments([text]);

  return docs.map((doc, index) => ({
    content: doc.pageContent,
    order: index
  }));
}

File: backend/src/utils/chunker.py
from langchain_text_splitters import RecursiveCharacterTextSplitter


def chunk_text(text: str) -> list[dict]:
    splitter = RecursiveCharacterTextSplitter(
        chunk_size=500,
        chunk_overlap=100,
        separators=["\n\n", "\n", " ", ""],
    )
    docs = splitter.create_documents([text])
    return [{"content": doc.page_content, "order": index} for index, doc in enumerate(docs)]
